h5py/_hl/files.py

```diff
@@ -502,6 +582,9 @@ def __init__(self, name, mode='r', driver=None, libver=None, userblock_size=None
             Passed on to the selected file driver.
         """
         if driver == 'ros3':
+            if not isinstance(name, (str, bytes)):
+                raise TypeError(f"with {driver=!r}, name must be str or bytes, not {type(name).__name__}")
```

Comment:
This will directly conflict with https://github.com/h5py/h5py/pull/2636, but an easy way out is to integrate it directly there. Ping @ajelenak